### PR TITLE
Updated prune_ecr_images to correctly handle private ECR repos.

### PR DESCRIPTION
**Pull request by barankyle**
## Summary
Converted that script to .ts

Made findBuilderTags filter on tagged status.

## References
closes #_insert number here_

### CODE REVIEW

File: scripts/prune_ecr_images.ts [hunk 1]
```diff
@@ -41,88 +47,127 @@ const options = cli.parse({
...
         )
       }
       currentImages = [...new Set(currentImages)]
     }
-    const ecr = (
+
+    console.log('EKS config', config.aws.eks)
+    const awsCredentials = `[default]\naws_access_key_id=${config.aws.eks.accessKeyId}\naws_secret_access_key=${config.aws.eks.secretAccessKey}\n[role]\nrole_arn = ${config.aws.eks.roleArn}\nsource_profile = default`
+
+    console.log('awsCredentials', awsCredentials)
```
Comment (lucas3900) [inline scripts/prune_ecr_images.ts]: If we don't have log level control, we shouldn't log the AWS access credentials to the console. And maybe `config.aws.eks` depending on what is stored there.

File: scripts/build_docker_builder.sh [hunk 2]
```diff
@@ -50,8 +50,8 @@ if [ $REPO_PROVIDER === "aws" ]
 then
   if [ $PRIVATE_REPO == "true" ]
   then
-    node ./scripts/prune_ecr_images.js --repoName $REPO_NAME-builder --region $REGION --service builder --releaseName $STAGE
+    npx ts-node ./scripts/prune_ecr_images.ts --repoName $REPO_NAME-builder --region $REGION --service builder --releaseName $STAGE
   else
-    node ./scripts/prune_ecr_images.js --repoName $REPO_NAME-builder --region us-east-1 --service builder --releaseName $STAGE --public
+    npx ts-node ./scripts/prune_ecr_images.ts --repoName $REPO_NAME-builder --region us-east-1 --service builder --releaseName $STAGE --public
```
Comment (lucas3900) [inline scripts/build_docker_builder.sh:55]: Public repos can be in any region. Should we use `$REGION` here as well?

File: scripts/build_and_publish_package.sh [hunk 3]
```diff
@@ -128,9 +128,9 @@ fi
 
 if [ $PRIVATE_REPO == "true" ]
 then
-  node ./scripts/prune_ecr_images.js --repoName $DESTINATION_REPO_NAME_STEM-$PACKAGE --region $REGION --service $PACKAGE --releaseName $STAGE
+  npx ts-node ./scripts/prune_ecr_images.ts --repoName $DESTINATION_REPO_NAME_STEM-$PACKAGE --region $REGION --service $PACKAGE --releaseName $STAGE
 else
-  node ./scripts/prune_ecr_images.js --repoName $DESTINATION_REPO_NAME_STEM-$PACKAGE --region us-east-1 --service $PACKAGE --releaseName $STAGE --public
+  npx ts-node ./scripts/prune_ecr_images.ts --repoName $DESTINATION_REPO_NAME_STEM-$PACKAGE --region us-east-1 --service $PACKAGE --releaseName $STAGE --public
```
Comment (lucas3900) [inline scripts/build_and_publish_package.sh:133]: Public repos can be in any region. Should we use `$REGION` here as well?

File: scripts/prune_ecr_images.ts [hunk 4]
```diff
@@ -41,88 +47,127 @@ const options = cli.parse({
...
-        ? new ECRPUBLICClient({ region: 'us-east-1' })
-        : new ECRClient({ region: options.region || 'us-east-1' })
-    )
-    const images = await getAllImages(ecr, options.repoName || 'etherealengine', undefined, [])
+        ? new ECRPUBLICClient({
+            credentials: fromIni({
+              profile: config.aws.eks.roleArn ? 'role' : 'default',
+              filepath: credentialsPath
+            }),
+            region: 'us-east-1'
```
Comment (lucas3900) [inline scripts/prune_ecr_images.ts:154]: here for public region as well

**Comment by lucas3900** [comment]
Wow the way the TS SDK makes your jump through hoops just to sort and delete on an ECR is insane.

File: scripts/prune_ecr_images.ts (hunk 1 above)
Comment (barankyle) [reply scripts/prune_ecr_images.ts]: These were debugging statements, I'll remove them when it's 100% ready.

File: scripts/build_docker_builder.sh (hunk 2 above)
Comment (barankyle) [reply scripts/build_docker_builder.sh:55]: I swear that every time I've tried to access a public repo that's located in a non-us-east-1 region using that region, it hasn't worked, but that using us-east-1 as the region has worked. I will re-check this to confirm.

Comment (barankyle) [reply scripts/build_docker_builder.sh:55]: Yeah, the SDK for ECR Public works a little weird when combined with how public repos are addressed. I think public repos are region-less, or are always in us-east-1; if I switch regions in the web console when looking at public repos, I always see every one.

If I set up ECRPUBLICClient with a region other than us-east-1, then try to call a DescribeImagesCommand, I get e.g. 
```
Uncaught Error: getaddrinfo ENOTFOUND api.ecr-public.us-west-1.amazonaws.com
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)
    at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:128:17) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'api.ecr-public.us-west-1.amazonaws.com',
  '$metadata': { attempts: 1, totalRetryDelay: 0 }
}
```
I don't know why it lets you change the region if anything other than us-east-1 is invalid, but that's why I hard-coded it to us-east-1.

Comment (lucas3900) [reply scripts/build_docker_builder.sh:55]: that's very weird. always fun exploring the hidden functionality of AWS

**Review by lucas3900 — APPROVED** [review]
Approved. Just leaving a note here to make sure we delete those logging statements. Might be good to set up a flow where we test those sort of AWS/K8s specific things on QAT, prior to merging into dev?

**Comment by barankyle** [comment]
> Approved. Just leaving a note here to make sure we delete those logging statements. Might be good to set up a flow where we test those sort of AWS/K8s specific things on QAT, prior to merging into dev?

I'd already deleted them. It is very useful to test them on QAT, which I did for this, making it explicit would be a good thing.

**Pull request merged**